crates/flux-fhir-analysis/src/wf/mod.rs

```diff
@@ -445,15 +488,20 @@ impl<'a, 'tcx> Wf<'a, 'tcx> {
                     ));
                 }
                 iter::zip(&path.refine, sorts)
-                    .try_for_each_exhaust(|(arg, sort)| self.check_refine_arg(infcx, arg, sort))?;
+                    .try_for_each_exhaust(|(arg, sort)| self.check_refine_arg(infcx, arg, &sort))?;
             }
             fhir::Res::SelfTyParam { .. }
             | fhir::Res::SelfTyAlias { .. }
             | fhir::Res::Def(..)
             | fhir::Res::PrimTy(..) => {}
         }
         let snapshot = self.xi.snapshot();
-        let args = self.check_generic_args(infcx, &path.args);
+        let adt_id = match &path.res {
+            // TODO:Enums
+            fhir::Res::Def(DefKind::Struct, did) => Some(*did),
+            _ => None,
+        };
```

Comment:
`fhir::Res::PrimTy`, `fhir::Res::SelfTyAlias` and `SelfTyParam` should never have generic arguments, it's an instant error to have them, so we can skip calling `self.check_generic_args` in those cases. This assumption is verified in `annot_check` when zipping it against the rust signature.

Most `DefKind`s have generics (or cannot appear in a path resolution) so I would filter `DefKind::TyParam` and call `self.check_generic_args` on the rest of `fhir::Res::Def`

Reply:
done